### DISCUSSION TITLE
SI7021 working on an old version of Tasmota. But stopped working after upgrading to 13.1.0.

### DISCUSSION THREAD

**Discussion by st2000** [arendst/Tasmota#19523] [Support request]
I have been researching this.  I think support for the SI7021 I2C interface (directly interfacing with the SI7021 temperature / humidity sensor) was dropped int 5.11.0 20280107.  In that version of Tasmota support was added for a temperature / humidity sensor which contained an SI7021 chip but that chip was connected to a  EFM8BB10F2G which converted the interface from I2C to a 1-wire interface.

My Tasmota device has already been built and has operated for years in an inaccessible (hard to reach) location.  How can I restore the direct I2C ST7021 support?  I think I need to change some definitions and recompile Tasmota basic.  But what do I need to change?

-thanks

**Comment by barbudor** [comment]
I2c driver is still available in Tasmota as i2cdriver 9 (USE_HTU)
It is built in `tasmota-sensors` and `tasmota32`
No need for self compile

**Reply by st2000** [reply]
That worked!  Thanks.
For others, looks like the tasmota sensor distribution doesn't fit into the 4CH Pro's memory.  I had to load in the tasmota minimal distribution 1st then load in the tasmota sensor distribution.  I suspect to have to repeat this procedure in the future should I ever need to upgrade to a newer version of tasmota.

**Reply by sfromis** [reply]
You could just do OTA upgrade using the URL directly (the .gz variant if ESP8266). Then Tasmota will automatically use minimal if/when needed, or via safeboot if ESP32.

**Reply by st2000** [reply]
I did not know that.  OTA / URL did take longer.  I thought it was because I was pulling across the web.  But, likely, it was because tasmota downloaded minimal, installed it then downloaded the final copy of firmware.  Clever!!
